### DISCUSSION TITLE
What are the TRV compatible for `over_valve` VTherm ?

### DISCUSSION THREAD

**Discussion by jmcollin78** [jmcollin78/versatile_thermostat#631] [Q&A]
Hello,

I want to list here all TRV (Thermostatic valve) which are compatible with Vtherm.
Please post your configuration if you have one.

**Comment by jmcollin78** [comment]
Shelly TRV valve are compatible: https://www.shellyfrance.fr/shelly-trv/

**Reply by LapinouJuteux** [reply]
Hi jmcollin78,

I am trying to set up versatile_thermostat with my shelly TRV, and could follow all the installation steps, however now I am stuck, as I have two climate entities in parallel the one from the shelly integration, which I refer to in the setup of the vtherm, and the vtherm one. Do I need to disable anything on the shelly side, or will the two live happily together?
I can see that the valve number of the vtherm integration does set the valve on the shelly integration, should I just set the shelly climate to off, and let vtherm drive everything?

Hope I make sense :-)
Thanks
Yvan

**Reply by jmcollin78** [reply]
Hello @LapinouJuteux  (!!),

With Shelly TRV you should use the `over_valve` VTherm. I have develop this kind for this TRV and give directly the valve number entity as an underlying.

**Reply by LapinouJuteux** [reply]
Hi @jmcollin78 ,

Thanks for the quick reply, it is what I picked, but I wasn't sure about the initial steps, I went ahead with my assumption of turning off the climate entity provided by the shelly integration. 
I probably have missed it in the detailed instructions, but if it is not there, it might be worthwhile to add.

Yvan

**Comment by jmcollin78** [comment]
Sonoff TRVZB is now fully compatible: use the `over_climate` VTherm and use the self-regulation with 'direct control of the valve'.
Don't set the closing degree for Sonoff TRVB but set the opening degree and the calibration offset to have the best experience.

**Reply by stephensalex** [reply]
> Sonoff TRVZB is now fully compatible: use the `over_climate` VTherm and use the self-regulation with 'direct control of the valve'. Don't set the closing degree for Sonoff TRVB but set the opening degree and the calibration offset to have the best experience.

Just to check this, is it recommended to not select the closing degree entity in the valve regulation configuration screen? Should the closing degree entity be set to zero on the trv itself for optimal performance? Thanks

**Reply by jmcollin78** [reply]
yes, see https://github.com/jmcollin78/versatile_thermostat/blob/main/documentation/en/self-regulation.md#self-regulation-by-direct-valve-control

> Note: for Sonoff TRVZB you should not configure the "closing degree" parameter. This leads to a bug in the TRV and the hvac_action is no more working.

**Reply by jmcollin78** [reply]
Just don't configure it. Other let the value to 100% if I understand well (I don't have myself...)

**Reply by stephensalex** [reply]
Thank you

**Reply by dantrio** [reply]
Hi, I've got some Sonoff TRVZB's and just started testing out as well. Just to confirm, the valve closing degree entity on the origional TRV device (mines setup in ZHA) should be left at the default 100, and then versatile thermostat controls the entity for the opening degree once this is setup in the valve regulation config screen, and this will give the best results? Sorry if thats a silly question am just starting to use home assistant.

**Reply by sleveille4** [reply]
interesting,

i do have several trvzb and switched from the old vavle control to this new climate+valve mode a few days ago, wich is better (also removed the mirroring script made this fall, as you include the function now, really nice addition)
i just came here to report a potential bug i had thius evening where all VT climate seemed stuck in OFF mode (regardless of what i set them to, they reverted to "off") the others (over switch) were working as intended.
nothing in the logs.
checked everything else, direct order to the TRV worked, they were not on AUTO mode, all other entitys working etc.....
no explaination to be found
had to restart HA to restore function
then i found this mention about the closing degree.

i removed them from the config, could the issue i encountered be the bug you mention?

if yes, very nasty one given that everything worked fine for almost a (cold) week before they all just "froze" in off mode! (and yes, it was rather cold when i came home :( )

@dantrio, yes, the proper setting for "closing degree" should be 100%, as i tested before @jmcollin78 included this mode, setting closing to 100% then offset to lowest, and climate to max, allowed "over valve" to gain almost direct control by changing opening degree, but was not perfect and required tinkering and script, 

now this new mode does it all automatically 👍 

@jmcollin78 just sent you some well deserved beers :)

**Comment by tbrasser** [comment]
In addition to sonoff I also own Bosch, they expose direct valve control to zigbee2mqtt, but it's read-only in Home Assistant (should be fixable on z2m side)
